cmd/pj-rehearse/main.go

```diff
@@ -113,6 +115,9 @@ func rehearseMain() int {
 		return gracefulExit(o.noFail, misconfigurationOutput)
 	}
 
+	metrics := rehearse.NewMetrics(o.metricsPath)
+	defer metrics.Dump()
```

Comment:
if the `metricsPath` will be empty, we shouldn't gather metrics at all. right?

Reply:
Correct, `Dump` is a NOP if the path is empty, so it's safe to defer like this.

https://github.com/openshift/ci-operator-prowgen/pull/108/files#diff-66bd766f197dd6b959d77192fcde3dbeR90